File: dartglue/tool/templates/android/NativeViewManager.kt
package com.dartglue

import android.content.Context
import android.view.View
import android.widget.TextView
import android.widget.FrameLayout
import android.view.ViewGroup
import org.json.JSONObject

class NativeViewManager(private val context: Context) {
    private val viewCache = mutableMapOf<Long, View>()
    private var handleCounter: Long = 1
    val rootView = FrameLayout(context)

    fun createView(type: Int): Long {
        val view = when (type) {
            0 -> FrameLayout(context)  // View
            1 -> TextView(context)     // Text
            else -> throw IllegalArgumentException("Unknown view type: $type")
        }
        
        val handle = handleCounter++
        viewCache[handle] = view
        return handle
    }

    fun updateViewProps(handle: Long, propsJson: String) {
        val view = viewCache[handle] ?: return
        val props = JSONObject(propsJson)
        
        // Update common properties
        props.optDouble("width", -1.0).takeIf { it > 0 }?.let {
            view.layoutParams?.width = it.toInt()
        }
        
        props.optDouble("height", -1.0).takeIf { it > 0 }?.let {
            view.layoutParams?.height = it.toInt()
        }
        
        // Handle text properties
        if (view is TextView) {
            props.optString("text")?.let { view.text = it }
            props.optDouble("fontSize", -1.0).takeIf { it > 0 }?.let {
                view.textSize = it.toFloat()
            }
        }
    }
}

File: dartglue/tool/templates/ios/NativeViewManager.swift
import UIKit

class NativeViewManager {
    private var viewCache = [Int64: UIView]()
    private var handleCounter: Int64 = 1
    let rootView = UIView()

    func createView(_ type: Int32) -> Int64 {
        let view: UIView = {
            switch type {
            case 0: return UIView()
            case 1: return UILabel()
            default: return UIView()
            }
        }()
        
        let handle = handleCounter
        handleCounter += 1
        viewCache[handle] = view
        return handle
    }
    
    func updateViewProps(_ handle: Int64, _ propsJson: String) {
        guard let view = viewCache[handle],
              let data = propsJson.data(using: .utf8),
              let props = try? JSONSerialization.jsonObject(with: data) as? [String: Any] else {
            return
        }
        
        // Update common properties
        if let width = props["width"] as? CGFloat {
            view.frame.size.width = width
        }
        
        if let height = props["height"] as? CGFloat {
            view.frame.size.height = height
        }
        
        // Handle text properties
        if let label = view as? UILabel {
            if let text = props["text"] as? String {
                label.text = text
            }
            if let fontSize = props["fontSize"] as? CGFloat {
                label.font = .systemFont(ofSize: fontSize)
            }
        }
    }
}